### DISCUSSION TITLE
using Strip() in .net

### DISCUSSION THREAD

**Discussion by duffypaul** [dlemstra/Magick.NET#865] [Help]
I am trying to replicate this command line in .net:
magick convert -units pixelsperinch -density 150 Filename.pdf[1] -strip
Filename_out.bmp

In particular the -strip part. Here's what I've got:

var _MagickReadSettings = new MagickReadSettings();
_MagickReadSettings.Density = new Density(150, DensityUnit.PixelsPerInch);
var img = new MagickImage("Filename.pdf[1]", _MagickReadSettings);
img.Strip();
img.Write(Filename_out.bmp);

I'm showing .bmp but tried other formats too. 

After stripping with the command line version:
magick identify Filename_out.bmp
(returns nothing, as expected)

After stripping from .net version:
magick identify Filename_out.bmp
Returns:
Filename_out.bmp BMP 595x842 595x842+0+0 8-bit sRGB 1.91126MiB 0.000u 0:00.006

It seems like the Strip() part did nothing... I'm very new to ImageMagick so if someone could point me in the right direction on this I'd be very grateful :)

Thanks!

**Comment by dlemstra** [comment]
The `-strip` option or `.Strip()` method removes metadata from the file. And I am not sure what you are doing on the command line but `magick identify` should always print some kind of result instead of nothing. Can you share your `PDF` file and the exact commands that you are executing?

**Reply by duffypaul** [reply]
Hi, thanks much for replying. I cannot share the pdf unfortunately. I'm basically trying to strip the metadata so tesseract can read the output image. Without doing a strip first I get this type of error message I'm sure you've seen before:

Warning: Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 98

If I use strip from command line with the code I posted ("out1"), this goes away and the output of tessearact is correct. If I use .net strip() method ("out2"), it doesn't seem to have any effect - still get the tesseract warning and the xml output is incorrect dimensions. 

I tried using magick identify with and without the -format option, and only without -format it returns anything (windows 10, command prompt). When ran on "out1" above it returns nothing. When ran on "out2" above it returns some metadata. It seems like Strip() in .net is doing nothing (?)

Using Magick.NET-Q16-AnyCPU v7.23.2
Visual Studio 2019
Windows 10

**Comment by dlemstra** [comment]
I really cannot help you without a complete example. The same code is being executing on the command line and in the library.

**Reply by duffypaul** [reply]
That's what I thought, but for some reason the output from the code is not the same. Unfortunately I can't provide the pdf as it's sensitive info from a customer. Shame but ok I'll find a different way. Thanks.

**Reply by dlemstra** [reply]
If you happen to find a random PDF document that can be used to reproduce this issue then please respond here.

**Reply by duffypaul** [reply]
Will try to reproduce later with sample data I can send, thanks a million.

**Reply by duffypaul** [reply]
Hey, sorry for the delay I was super busy. I didn't exactly reproduce the error with another pdf, I only got it to show a difference by running the output images through tesseract. With the cmd line strip, it has no problem. But with the .net strip() tesseract complains about invalid resolutions as in the previous example. Do you want me to send the test code for this case?

**Reply by dlemstra** [reply]
At this point I don't think I have enough information. Would it be possible to create a small Program.cs file that demonstrates the problem? And show me the ImageMagick commands that you are running and the output of those commands?
